### PR TITLE
Refine what a relative uri means in a git path

### PR DESCRIPTION
**Pull request by sigurdm**

### CODE REVIEW

File: lib/src/source/git.dart [hunk 1]
```diff
@@ -147,9 +147,15 @@ class GitSource extends Source {
     return {'relative': relative, 'url': url};
   }
 
-  /// Returns [path] normalized.
+  /// Normalizes [path].
   ///
-  /// Throws a [FormatException] if [path] isn't a relative url or null.
+  /// Throws a [FormatException] if [path] isn't a String parsing as arelative
```
Comment (lrhn) [inline lib/src/source/git.dart:152]: String -> [String]
arelative -> a relative

File: lib/src/source/git.dart [hunk 2]
```diff
@@ -147,9 +147,15 @@ class GitSource extends Source {
...
+  /// url or null.
```
Comment (lrhn) [inline lib/src/source/git.dart:153]: Ob-pedantry: 

url -> URL
null -> `null`

File: lib/src/source/git.dart [hunk 3]
```diff
@@ -159,9 +165,13 @@ class GitSource extends Source {
...
     final parsed = Uri.parse(path);
-    if (parsed.isAbsolute) {
+    if (parsed.hasAbsolutePath ||
+        parsed.hasScheme ||
+        parsed.hasAuthority ||
+        parsed.hasFragment ||
+        parsed.hasQuery) {
       throw FormatException(
-          "The 'path' field of the description must be relative.");
+          "The 'path' field of the description must be a relative path url.");
```
Comment (lrhn) [inline lib/src/source/git.dart:174]: url -> URL

**Review by lrhn — APPROVED** [review]

**Review by szakarias — APPROVED** [review]

File: lib/src/source/git.dart (hunk 1 above)
Comment (sigurdm) [reply lib/src/source/git.dart:152]: Done

File: lib/src/source/git.dart (hunk 2 above)
Comment (sigurdm) [reply lib/src/source/git.dart:153]: Done

File: lib/src/source/git.dart (hunk 3 above)
Comment (sigurdm) [reply lib/src/source/git.dart:174]: Done

**Pull request merged**